rubix-spi/src/main/java/com/qubole/rubix/spi/CacheUtil.java

```diff
@@ -244,13 +256,33 @@ private static String getDirectory(String remotePath, Configuration conf)
         relLocation = relLocation.substring(1);
       }
     }
-
+    if (CacheConfig.isPathEncryptionEnabled(conf)) {
+      relLocation = hashedPaths.getUnchecked(relLocation);
+    }
     final String absLocation = getLocalDirFor(remotePath, conf) + relLocation;
     createCacheDirectory(absLocation);
 
     return absLocation;
   }
 
+  private static String getHashedPath(String relLocation)
+  {
+    String hashRelLocation = relLocation;
+    try {
+      MessageDigest md = MessageDigest.getInstance("SHA-256");
+      byte[] pathBytes = md.digest(relLocation.getBytes());
+      StringBuilder sb = new StringBuilder();
+      for (int i = 0; i < pathBytes.length; i++) {
+        sb.append(Integer.toString((pathBytes[i] & 0xff) + 0x100, 16).substring(1));
+      }
+      hashRelLocation = sb.toString();
+    }
+    catch (NoSuchAlgorithmException e) {
+      log.error("No Such Algorithm for Hashing " + e.toString());
```

Comment:
`log.error()` can take an exception as a second parameter. This is preferred, rather than concatenating the exception as a String.

Reply:
Done